### PR TITLE
feat(session-search): add message position (offset / total) to search results

### PR DESCRIPTION
**Pull request by sgfvamll**
Follow-up to #4648

## Summary

`session_search` results now include a `position: offset / total` line for each match, showing the message's 0-based index within its session and the total message count. This enables precise lookups via `session_read(session_id, offset=<offset>, limit=1)` without needing `session_message_read`.

## Changes

- **types.ts**: Add `message_offset: number` and `total_messages: number` to `SearchResult`
- **session-formatter.ts**: `searchInSession` computes offset via indexed loop; `formatSearchResults` renders `position: offset / total`
- **constants.ts**: Update `SESSION_SEARCH_DESCRIPTION` example output with position lines
- **session-formatter.test.ts** (new): 14 tests covering offset computation, total count, multi-match distinctness, empty/non-existent sessions, and position rendering

## Example output

```
Found 3 matches across 2 sessions:

[ses_abc123] Message msg_001 (user)
  position: 0 / 45
  ...implement the **session manager** tool...
  Matches: 1

[ses_def456] Message msg_012 (user)
  position: 11 / 30
  ...use the **session manager** to find...
  Matches: 1
```

## Test

70 pass, 0 fail across 5 files in `src/tools/session-manager/`.

---
## Summary by cubic
Add message position (offset / total) to session search results so you can find a hit within its session and fetch it via `session_read(session_id, offset=N, limit=1)`. Each match now shows a "position: offset / total" line in the output.

- **New Features**
  - Extend `SearchResult` with `message_offset` and `total_messages`.
  - Compute offsets in `searchInSession` and render position in `formatSearchResults`.
  - Update `SESSION_SEARCH_DESCRIPTION` example and add tests for offsets, totals, empty sessions, and rendering.

<sup>Written for commit 32972ad141ac322e306fe28fd01f3ab270bcd197. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/code-yeongyu/oh-my-openagent/pull/4651?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: src/tools/session-manager/constants.ts [hunk 1]
```diff
@@ -50,16 +50,23 @@ Arguments:
 - case_sensitive (optional): Case-sensitive search (default: false)
```
[the remark this answers is not in the document]
Comment (sgfvamll) [reply src/tools/session-manager/constants.ts:50]: This pr follows up to https://github.com/code-yeongyu/oh-my-openagent/pull/4648, which adds the `offset` parameter.

**Comment by code-yeongyu** [comment]
[sisyphus-bot] Thanks for this feature, @sgfvamll. The session-manager directory has been restructured on dev (moved to packages/omo-opencode/src/tools/session-manager/), so this PR would need a rebase. We're closing this as stale. If message position in search results is still valuable, please open a new PR against the current dev structure.
